src/app/libraries/token_generator_lib.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+class token_generator_lib extends Library {
+
+    public function __construct() {
+        if (session_status() == PHP_SESSION_NONE) {
+            session_start();
+        }
+    }
+    /**
+     * Generates a new CSRF token
+     * @return string Base64 encoded token
+     */
+    public static function getNewCSRFToken() {
+        return base64_encode(openssl_random_pseudo_bytes(12));
+    }
+
+    /**
+     * Returns the currently set CSRF token
+     * @return string|bool The token
+     */
+    public function getCurrentCSRFToken() {
```

Comment:
This is what gets passed to the view as data. Instead of copying the token into a variable at setting time, I just pass along the value returned from this method.